Code/app/src/main/java/ca/ualberta/moodroid/ui/EditMoodDetail.java

```diff
@@ -0,0 +1,313 @@
+                } catch (Exception e) {
+                    Log.e("EDIT/DATE", "Issue setting the date from the mood..." + e.getMessage());
+                }
+
+                // setup the mood model
+                setInitialMoodEvent(moodEventModel);
+                //set social situation
+                String val = moodEventModel.getSituation();
+                if (!(val == null)) {
+                    if (val.equals("Alone")) {
+                        social_situation.setSelection(1);
+                    } else if (val.equals("One Other Person")) {
+                        social_situation.setSelection(2);
+                    } else if (val.equals("Two to Several People")) {
+                        social_situation.setSelection(3);
+                    } else if (val.equals("Crowd")) {
+                        social_situation.setSelection(4);
+                    }
+                }
+                //set location
+                if (moodEventModel.getLocation() != null){
+                    double latt = moodEventModel.getLocation().getLatitude();
+                    double longt = moodEventModel.getLocation().getLongitude();
+                    String loc = String.valueOf(latt) + ", " + String.valueOf(longt);
+                    locationText.setText(loc);
+                }
+                //set reason
+                if (moodEventModel.getReasonText().length() > 0) {
+                    reason_text.setText(moodEventModel.getReasonText());
+                }
+                if (moodEventModel.getReasonImageUrl() != null) {
+                    try {
+                        Glide.with(EditMoodDetail.this)
+                                .load(moodEventModel.getReasonImageUrl())
+                                .into(img);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        Toast.makeText(EditMoodDetail.this, "Error: Image cannot be displayed. " + moodEventModel.getReasonImageUrl(), Toast.LENGTH_SHORT).show();
+                    }
+                }
+
+            }
+        });
+
+    }
+
+    /**
+     * Initialize the text and edit views on the display here
+     */
+    protected void initializeViews() {
+        //initializing firebase storage
+        storage = FirebaseStorage.getInstance();
+        storageReference = storage.getReference();
+
+        //initializing TextWatcher to check for invalid reason text input
+        reason_text.addTextChangedListener(textWatcher);
+
+        // initializing the views that will be set from the last activity
+        img = findViewById(R.id.photoView);
+        mood_img = findViewById(R.id.mood_img);
+        mood_title = findViewById(R.id.mood_text);
+        banner = findViewById(R.id.banner);
+        social_situation.setAdapter(new ArrayAdapter<>(this, android.R.layout.simple_spinner_dropdown_item, EditMoodDetail.situations));
+        removePhotoButton.setVisibility(GONE);
```

Comment:
We probably need to check the state of location/photo before setting this visibility settings.